envoy-control-core/src/main/kotlin/pl/allegro/tech/servicemesh/envoycontrol/snapshot/resource/routes/ServiceTagMetadataGenerator.kt

```diff
@@ -1,16 +1,26 @@
 package pl.allegro.tech.servicemesh.envoycontrol.snapshot.resource.routes
 
-import com.google.re2j.Pattern
 import pl.allegro.tech.servicemesh.envoycontrol.snapshot.ServiceTagsProperties
 
 /**
  * Used for cluster endpoints to provide subset routing
  */
 class ServiceTagMetadataGenerator(properties: ServiceTagsProperties = ServiceTagsProperties()) {
 
-    private val tagsBlacklist: Pattern = properties.routingExcludedTags
-            .joinToString("|")
-            .let { Pattern.compile(it) }
+    data class ListTag(val tagMatcher: String, val type: ListTagType)
+
+    enum class ListTagType {
+        EXACT, PREFIX
+    }
+
+    private val tagsBlacklist = properties.routingExcludedTags.map {
```

Comment:
I would add 3 different types: `prefix`, `regex`, `exact`. Instead of making magic in code I would make properties which describes how each pattern should be resolved.
```
routing-excluded-tags:
  prefix: "abc"
  regex: ".*"
  exact: "abc"
```